src/Nlog.RabbitMQ.Target/RabbitMQTarget.cs

```diff
@@ -334,35 +334,45 @@ private byte[] GetMessage(LogEventInfo logEvent)
 		}
 
 		private string GetMessageString(LogEventInfo logEvent)
-		{
-			if (!UseJSON)
-			{
-				return RenderLogEvent(Layout, logEvent);
-			}
-			else
-			{
-				var message = this.UseLayoutAsMessage ? RenderLogEvent(Layout, logEvent) : logEvent.FormattedMessage;
-				var messageSource = RenderLogEvent(MessageSource, logEvent);
-				if (string.IsNullOrEmpty(messageSource))
-					messageSource = string.Format("nlog://{0}/{1}", System.Net.Dns.GetHostName(), logEvent.LoggerName);
-
-				try
-				{
-					var jsonSerializer = JsonSerializer;
-					lock (jsonSerializer)
-					{
-						return MessageFormatter.GetMessageInner(jsonSerializer, message, messageSource, logEvent, this.Fields);
-					}
-				}
-				catch (Exception e)
-				{
-					_jsonSerializer = null;	// reset as it it might be broken
-					InternalLogger.Error(e, "RabbitMQTarget(Name={0}): Failed to serialize LogEvent: {1}", Name, e.Message);
-					throw;
-				}
-			}
+		{
+			if (!UseJSON)
+			{
+				return RenderLogEvent(Layout, logEvent);
+			}
+			else
+			{
+				var message = this.UseLayoutAsMessage ? RenderLogEvent(Layout, logEvent) : logEvent.FormattedMessage;
+				var messageSource = RenderLogEvent(MessageSource, logEvent);
+				if (string.IsNullOrEmpty(messageSource))
+					messageSource = string.Format("nlog://{0}/{1}", System.Net.Dns.GetHostName(), logEvent.LoggerName);
+
+				try
+				{
+					var jsonSerializer = JsonSerializer;
+					lock (jsonSerializer)
+                    {
+						return MessageFormatter.GetMessageInner(jsonSerializer, message, messageSource, logEvent, Fields, GetFullContextProperties(logEvent));
```

Comment:
Can you move the call GetFullContextProperties(logEvent) out of the try-catch, so it happens just after `var message = `

```c#
var message = this.UseLayoutAsMessage ? RenderLogEvent(Layout, logEvent) : logEvent.FormattedMessage;
var contextProperties = GetFullContextProperties(logEvent);
```